src/net/gateway-pool.ts

```diff
@@ -90,6 +90,18 @@ export class GatewayPool implements IGatewayClient {
         return this.activeGateway.isReady
     }
 
+    public get archiveIpnsId(): string {
+        return this.activeGateway.archiveIpnsId
+    }
+
+    public set archiveIpnsId(ipnsId: string) {
```

Comment:
Why do we need to get the complete list of different IPNS ID's?

Theoretically, we only care about the active gateway... but even if we also cared about the rest, they all should have been initialized with the same IPNS ID